packages/millicast-sdk/src/View.js

```diff
@@ -247,31 +251,36 @@ export default class View extends BaseWebRTC {
     this.stopReemitingWebRTCPeerInstanceEvents = reemit(webRTCPeerInstance, this, Object.values(webRTCEvents))
     this.stopReemitingSignalingInstanceEvents = reemit(signalingInstance, this, [signalingEvents.broadcastEvent])
 
-    const workerBlob = new Blob([workerString])
-    const workerURL = URL.createObjectURL(workerBlob)
-    this.worker = new Worker(workerURL)
-
-    webRTCPeerInstance.on('track', (trackEvent) => {
-      this.tracksMidValues[trackEvent.transceiver?.mid] = trackEvent.track
-      if (supportsRTCRtpScriptTransform) {
-        // eslint-disable-next-line no-undef
-        trackEvent.receiver.transform = new RTCRtpScriptTransform(this.worker, {
-          name: 'receiverTransform',
-          payloadTypeCodec: { ...this.payloadTypeCodec },
-          codec: this.options.codec,
-          mid: trackEvent.transceiver?.mid
-        })
-      } else if (supportsInsertableStreams) {
-        const { readable, writable } = trackEvent.receiver.createEncodedStreams()
-        this.worker.postMessage({
-          action: 'insertable-streams-receiver',
-          payloadTypeCodec: { ...this.payloadTypeCodec },
-          codec: this.options.codec,
-          mid: trackEvent.transceiver?.mid,
-          readable,
-          writable
-        }, [readable, writable])
+    if (this.options.metadata) {
+      const workerBlob = new Blob([workerString])
+      const workerURL = URL.createObjectURL(workerBlob)
+      if (!this.worker) {
+        this.worker = new Worker(workerURL)
```

Comment:
Should we clean up the `worker` object in case it was set previously?

Reply:
If the worker is set previously, we could reuse it right? In the `stop` method we do the cleanup. If it is thought that the worker should be replaced in those cases we can clean up as it was. This was change to avoid creating a new Worker each time the method is called.

Reply:
If you reconnect to a stream without the need for metadata, then you want to clean up the worker to release the resources.
This method is only called when you connect to a new stream, so spending a little bit more time (few ms) on initialization should not be a big deal IMO.

Reply:
When reconnecting, it first calls `stop` (so worker terminated) and then starts connection process again, so I don't know if it's necessary to do a terminate there, right?

Reply:
Ok, if the `stop` is called all the time, then we're good with not cleaning up the worker.